src/DotVVM.Framework/Resources/Scripts/DotVVM.ts

```diff
@@ -391,7 +397,12 @@ class DotVVM {
             this.postJSON(<string>this.viewModels[viewModelName].url, "POST", ko.toJSON(data), response => {
                 try {
                     this.isViewModelUpdating = true;
-                    const result = JSON.parse(response.responseText);
+                    const responseJson = <IDotvvmStaticCommandResponse>JSON.parse(response.responseText);
+                    const result = responseJson.result;
+                    if (responseJson.action === "redirect") {
+                        // redirect
+                        this.handleRedirect(responseJson, viewModelName)
+                    }
```

Comment:
It's unclear whether the method should continue execution in case of a redirect. When the redirect happens on a "normal page", it will abort the execution, since the page uploads (I think, haven't tested). If we are performing SPA redirect, the `this.handleRedirect` method returns and the flow continues.

* It is inconsistent with `postbackCore` which interrupts the method (by returning and resolving the promise).
* It will try to finish the rest of the static command, which may starts another request racing with the navigation.

Reply:
I think it should not continue with the method execution and it should `reject` (although normal postbackCore `resolve`s :/ )

Reply:
Why we should reject it? If we want to be consistent, I would much more rather `resolve` it as postbackCore does.

Reply:
Well, yes, but there will be (probably) nothing happening after postbackCore. When we `resolve` here, the static command execution will continue, which may cause race conditions and strange behavior in SPA pages.

Reply:
Ok, I will add the return statement.

Reply:
Only the return statement is also not the best since it makes the promise run forever.

Reply:
Could you please reject it by calling `error`.

Reply:
Do you mean `errorCallback({ xhr: response, error: null });`?

Reply:
Oh, I see. You have already changed it.

Reply:
I have only changed the type. But I have changed that now - the promise gets rejected when redirect should happen.